### PR TITLE
reorganize the machineconfigpools package

### PR DESCRIPTION
**Pull request by ffromani**
We are preparing for a new API version, and the `machineconfigpools` package is getting in the way.
This package never had a clear design and was more a collection of utility functions, which are also closely coupled with the API.
This PR wants to de-entangle this mess, clean up and rationalize the design, and make room for easier introduction of new API versions in the near future.

### CODE REVIEW

**Comment by ffromani** [comment]
`CI build` lane failed for quota exceeded, not blocking

**Comment by ffromani** [comment]
`ci-e2e` lane also failed for infra issue (:\)

File: api/numaresourcesoperator/v1alpha1/helper/nodegroup/nodegroup.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		}
+
+		tree := Tree{
+			NodeGroup: nodeGroup,
+		}
+
+		for i := range mcps.Items {
+			mcp := &mcps.Items[i]
+
+			selector, err := metav1.LabelSelectorAsSelector(nodeGroup.MachineConfigPoolSelector)
```
Comment (jlojosnegros) [inline api/numaresourcesoperator/v1alpha1/helper/nodegroup/nodegroup.go:64]: Being this code only dependant of `nodeGroup` should not it be in the outside for loop?

**Review by jlojosnegros — APPROVED** [review]
/lgtm

**Pull request merged**